### PR TITLE
switch to Green/Continuation for Nevanlinna

### PR DESCRIPTION
**Pull request by iskakoff**
Instead of using exchange via filesystem, new interface allows to work with numpy array directly, simplifying the code.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -11,3 +11,4 @@ cvxpy
 # Solvers for CVXPY
 clarabel
 mosek
+green-ac
```
Comment (iskakoff) [inline requirements.txt:14]:
```suggestion
green-ac

```

File: green_mbtools/pesto/analyt_cont.py [hunk 2]
```diff
@@ -193,83 +193,11 @@ def nevan_run(
...
-                )
-        X_w = X_w.reshape((freqs.shape[0],) + X_iw_shape[1:])
-        X_iw = X_iw.reshape(X_iw_shape)
-    else:
-        print("All AC fails. Will not dump to DOS.h5")
-        X_iw = X_iw.reshape(X_iw_shape)
-
-    os.chdir(wkdir)
+    freqs = np.linspace(w_min, w_max, n_real)
+    X_w = green_ac.solve("Nevanlinna", wsample*1.j, freqs +1.j * eta, X_iw, prec)
```
Comment (gauravharsha) [inline green_mbtools/pesto/analyt_cont.py:197]: Linting suggestion: I'd make it either {"freqs*1.j"} or {"wsample * 1.j" and "freqs + 1.j"} to comply with PEP8 standard E225 which says "there should be a space before and after all operators".
The second option is used in almost all PESTO scripts.

File: green_mbtools/pesto/analyt_cont.py [hunk 3]
```diff
@@ -8,7 +8,7 @@
 from .ac_utils import dump_input_caratheodory_data, load_caratheodory_data
 import green_mbtools.pesto.nevanlinna as nevan_exe
```
Comment (gauravharsha) [inline green_mbtools/pesto/analyt_cont.py:9]: We should remove the unused import statement here. Also, we might as well remove the entire Nevanlinna code from the repository.

**Review by gauravharsha — COMMENTED** [review]
Just a few questions before we merge it in:
1. Is the parallelization scheme of green_ac same as the nevan_exe?
2. All the unit tests (for Dirac delta functions) are passing. But I think we should ensure the two implementations give identical (or similar?) results for realistic systems, e.g., H2 or silicon.

File: green_mbtools/pesto/analyt_cont.py (hunk 2 above)
Comment (iskakoff) [reply green_mbtools/pesto/analyt_cont.py:197]: I am thinking even to slightly change interface and pass only real part of frequencies, and for real frequency provide `eta` as an additional parameter, or probably as an element in `kwargs`.

File: green_mbtools/pesto/analyt_cont.py (hunk 3 above)
Comment (iskakoff) [reply green_mbtools/pesto/analyt_cont.py:9]: agree

**Comment by iskakoff** [comment]
> Just a few questions before we merge it in:
> 1. Is the parallelization scheme of green_ac same as the nevan_exe?

Parallelization is done different:
  - it uses `ProcessPoolExecutor` to spawn workers instead of making own pool of processes, which in some sense better for parallelization as it does not require independent workers to wait until pool is empty.
  - number of workers is chosen similarly as before (either takes it from slurm's environment variable or uses half of available cores)
  - results directly go into a memory rather than through filesystem (we no longer need to have all that massive number of generated files)

> 2. All the unit tests (for Dirac delta functions) are passing. But I think we should ensure the two implementations give identical (or similar?) results for realistic systems, e.g., H2 or silicon.
  - I think that has to be tested with green-ac itself
  - It is for sure will be different, for complicated systems Nevanlinna is extremely susceptible to small roundoff noise, old code uses text files to exchange data and that always comes with some additional roundoff.
  - I will make a test that uses same input for both implementations and use old code's results as a "reference" data to compare.

File: green_mbtools/pesto/analyt_cont.py (hunk 2 above)
Comment (gauravharsha) [reply green_mbtools/pesto/analyt_cont.py:197]: Agreed, that would be cleaner

**Comment by gauravharsha** [comment]
@iskakoff thanks! an extra test might be helpful. Even with the numerical noise generally present in molecular or solids data, I am positive that we should get very good agreement for spectral features near the Fermi level, e.g., the HOMO (or VBMax) quasiparticle peak.

**Review by vibinabraham — APPROVED** [review]

File: green_mbtools/pesto/analyt_cont.py
Comment (gauravharsha) [inline green_mbtools/pesto/analyt_cont.py:1]: one last thing that I missed in the first review: the kwarg `outdir` is no longer required in the signature of `nevan_run` function.

**Review by gauravharsha — CHANGES_REQUESTED** [review]
can merge after we adjust the function signature and calls for `nevan_run`.

File: green_mbtools/pesto/analyt_cont.py
Comment (iskakoff) [reply green_mbtools/pesto/analyt_cont.py:1]: I suggest to keep `outdir` parameter for `AC_nevanlinna` function but print a deprecation warning. As I can see all examples use it and, I believe, custom scripts that other people have also use it.

**Pull request merged**